static/js/tools/choropleth/choropleth.tsx

```diff
@@ -36,27 +43,40 @@ type StateType = {
 };
 
 class ChoroplethMap extends Component<PropsType, StateType> {
-  public state = {
-    geoJson: [] as any,
-    data: {},
-    date: "latest",
-    mapContent: {} as any,
-    pc: false,
-    popMap: {},
-    values: {},
-  };
-
-  public componentDidMount = (): void => {
+  svgContainerElement: React.RefObject<HTMLDivElement>;
+  constructor(props: PropsType) {
+    super(props);
+    this.svgContainerElement = React.createRef();
     const urlParams = new URLSearchParams(window.location.search);
     const isPerCapita =
       urlParams.has("pc") &&
       ["t", "true", "1"].includes(urlParams.get("pc").toLowerCase());
-    this.setState({
+    this.state = {
+      geoJson: [] as any,
```

Comment:
why here needs "as any"?

Reply:
I was just moving the original code, but I can change it